hrjobcontract/CRM/Hrjobcontract/Upgrader.php

```diff
@@ -1065,41 +1065,91 @@ public function createInsurancePlanTypes() {
   }
 
   /**
-   * Changes Pension Provider to be a contact, instead of an option_value.
+   * Changes Pension Provider to be a contact, instead of an option_value.  All
+   * required changes are wrapped in a transaction and rolled-back if any 
+   * problems arise.
    * 
-   * @return true
+   * @return boolean
+   *   true on success
+   * 
+   * @throws Exception
    */
   function upgrade_1026() {
     $messages = [];
     $pensionsTableName = CRM_Hrjobcontract_BAO_HRJobPension::getTableName();
 
-    $this->createPensionProviderType($messages);
-    $this->replacePensionProviders($pensionsTableName, $messages);
-    $this->removePensionTypeOptionGroup($messages);
-
-    CRM_Core_DAO::executeQuery("
-      ALTER TABLE {$pensionsTableName} 
-      MODIFY pension_type INT(10) unsigned,
-      ADD CONSTRAINT pension_type_contact_id_fk
-      FOREIGN KEY(pension_type)
-      REFERENCES civicrm_contact(id)
-      ON DELETE SET NULL ON UPDATE CASCADE;
-    ");
+    $tx = new CRM_Core_Transaction();
+    try {
+      $this->createPensionProviderType($messages);
+      $this->replacePensionProviders($pensionsTableName, $messages);
+      $this->removePensionTypeOptionGroup($messages);
+      $this->alterPensionsTable($pensionsTableName, $messages);
+    } catch (Exception $ex) {
+      $tx->rollback();
+      throw new Exception('Cannot do upgrade - ' . $ex->getMessage());
+    }
 
     if (!empty($messages) && function_exists('drupal_set_message')) {
```

Comment:
Oh, I see. I this case, do you think the warning will make any difference? Think you're the user updating the extension, then the upgrade runs successfully but you will still see those warnings. So what? Should I be worried? Should I do something about it? Is everything really fine? I'm afraid it will just make the users more confused. If everything went fine, I don't think we should warn them about anything.